mapfishapp/src/main/webapp/app/js/GEOR_Querier.js

```diff
@@ -0,0 +1,330 @@
+                    displayField: "name",
+                    listWidth: 165,
+                    tpl: GEOR.util.getAttributesComboTpl()
+                }
+            },
+            attributes: this.attributeStore,
+            toolbarType: "tbar",
+            map: this.map,
+            allowGroups: true,
+            noConditionOnInit: false,
+            deactivable: true,
+            autoScroll: true,
+            allowSpatial: true,
+            stateProvider: Ext.state.Manager.getProvider(),
+            vectorLayer: this.layer
+        });
+        this.items = [this.filterbuilder];
+
+        this.buttons = [{
+            text: OpenLayers.i18n("Close"),
+            handler: this.close,
+            scope: this
+        }, {
+            text: OpenLayers.i18n("Search"),
+            handler: this.search,
+            scope: this
+        }];
+
+        this.addEvents(
+            /**
+             * @event searchresults
+             * Fires when we've received a response from server 
+             *
+             * Listener arguments:
+             * options - {Object} A hash containing response, model and format
+             */
+            "searchresults",
+
+            /**
+             * @event search
+             * Fires when the user presses the search button
+             *
+             * Listener arguments:
+             * panelCfg - {Object} Config object for a panel 
+             */
+            "search"
+        );
+
+        GEOR.Querier.superclass.initComponent.call(this);
+    },
+
+
+    /**
+     * Method: _checkFilter
+     * Checks that a filter is not missing items.
+     *
+     * Parameters:
+     * filter - {OpenLayers.Filter} the filter
+     *
+     * Returns:
+     * {Boolean} Filter is correct ?
+     */
+    _checkFilter: function(filter) {
+        var filters = filter.filters || [filter];
+        for (var i=0, l=filters.length; i<l; i++) {
+            var f = filters[i];
+            if (f.CLASS_NAME == "OpenLayers.Filter.Logical") {
+                if (!this._checkFilter(f)) {
+                    return false;
+                }
+            } else if (!(f.value && f.type && 
+                (f.property || f.CLASS_NAME == "OpenLayers.Filter.Spatial"))) {
+                GEOR.util.infoDialog({
+                    msg: tr("Fields of filters with a red mark are mandatory")
+                });
+                return false;
+            }
+        }
+        return true;
+    },
+
+
+    /**
+     * Method: search
+     * Gets the Filter Encoding string and sends the getFeature request
+     */
+    search: function() {
+        // we quickly check if nothing lacks in filter
+        var filter = this.filterbuilder.getFilter();
+        if (!this._checkFilter(filter)) {
+            return;
+        }
+
+        this.fireEvent("search", {
+            // TODO: tell on which layerRecord the search is running ?
+            html: tr("<div>Searching...</div>")
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["tr" is not defined.](https://www.codacy.com/app/georchestra/georchestra/file/2338282099/issues/source?bid=3200181&fileBranchId=3209168#l269)